store/iavlstore.go

```diff
@@ -161,6 +162,10 @@ func (s *IAVLStore) SaveVersion(opts *VersionedKVStoreSaveOptions) ([]byte, int6
 	if flushInterval == 0 || ((oldVersion+1)%flushInterval == 0) {
 		if flushInterval != 0 {
 			log.Info("[IAVLStore] Flushing mem to disk", "version", oldVersion+1)
+			s.previousTree, err = s.tree.GetImmutable(oldVersion)
```

Comment:
Probably need to check that `oldVersion > 0` first.
Also `s.previousTree` must be accessed atomically, similarly to the `lastSavedTree` in the MultiWriterAppStore